Fix Navigation Bar Buttons Not Visible with Signup Sheet

ashiagr opened the pull request:
This PR tries to fix  #10908

It is an existing [Android issue](https://issuetracker.google.com/issues/65477640) which has become obsolete because of lower priority.  Seems like we need to go for a hack to fix it. Tried different [solutions](https://stackoverflow.com/questions/47553936/prevent-bottomsheetdialogfragment-covering-navigation-bar), this one seems to be working well in both portrait and landscape mode.

### To test:
1. Clear app data.
2. Launch app.
3. Tap Sign Up for WordPress.com button.
4. Notice navigation buttons are visible.

| API | Orientation | Before | After |
| ------ | ------------- | ------------- | ------------- |
| 27 | Portrait |  ![API27Before](https://user-images.githubusercontent.com/1405144/71882354-56205a80-315a-11ea-9b78-414405a129b4.png)|![API27After](https://user-images.githubusercontent.com/1405144/71882496-94b61500-315a-11ea-894b-5e9860a2dacc.png)|
| 27 | Landscape | ![API27BeforeLandscape](https://user-images.githubusercontent.com/1405144/71882742-0a21e580-315b-11ea-853b-a40785501f10.png)| ![API27AfterLandscape](https://user-images.githubusercontent.com/1405144/71882772-17d76b00-315b-11ea-922e-8503583bbb19.png)|
| 29 | Portrait | ![Android10BeforePortrait](https://user-images.githubusercontent.com/1405144/71882898-4a816380-315b-11ea-8c9b-d0774baf246b.png)|![Android10AfterPortrait](https://user-images.githubusercontent.com/1405144/71883051-992efd80-315b-11ea-9bfe-14b10b3ad8b8.png) |
| 29 | Landscape | ![Android10BeforeLandscape](https://user-images.githubusercontent.com/1405144/71883501-954fab00-315c-11ea-8132-b40b8ddb18be.png) | ![Android10AfterLandscape](https://user-images.githubusercontent.com/1405144/71883527-9e407c80-315c-11ea-8fcc-36aeb382ad05.png)|



PR submission checklist:

- [X] I have considered adding unit tests where possible.

- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

ashiagr commented:
> The changes included updating the base class from BottomSheetDialog to BottomSheetDialogFragment as well as updating styles.xml files.
Thank you for the directions @theck13 🙇‍♀️. I'll take a look and update it.

ashiagr commented:
Updated screenshots after applying changes requested.

| API | Portrait | Landscape |
| ------ | ------------- | ------------- |
| 26 |  ![api26_portrait](https://user-images.githubusercontent.com/1405144/72232742-0d98ef00-35e9-11ea-9e04-ab4068ce53d0.png) |  ![api26_land](https://user-images.githubusercontent.com/1405144/72232767-45079b80-35e9-11ea-897d-6e4e2066bf38.png) |
| 27 |  ![api27_portrait](https://user-images.githubusercontent.com/1405144/72232773-55b81180-35e9-11ea-927d-5fc366248c69.png) |  ![api27_land](https://user-images.githubusercontent.com/1405144/72232783-5fda1000-35e9-11ea-97d8-95b1edc52a5f.png) |
| 29 | ![api29_portrait](https://user-images.githubusercontent.com/1405144/72232803-7b451b00-35e9-11ea-96a0-bf9e690041d8.png) | ![api29_land](https://user-images.githubusercontent.com/1405144/72232810-8304bf80-35e9-11ea-86f3-8b16bd282b81.png) |

malinajirka commented on libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/SignupBottomSheetDialogFragment.java:
```diff
@@ -0,0 +1,100 @@
+            dialog.setOnDismissListener(new DialogInterface.OnDismissListener() {
+                @Override
+                public void onDismiss(DialogInterface dialog) {
+                    mSignupSheetListener.onSignupSheetCanceled();
```
It seems this call is duplicated - we invoke the same callback in `public void onDismiss(DialogInterface dialog)`. Is this intentional?

ashiagr replied:
`dialog.setOnDismissListener` was initially added referring to [SimpleNote dialog](https://github.com/Automattic/simplenote-android/blob/6978afe6744db9c5f0a3141cccc1d3aa01c86772/Simplenote/src/main/java/com/automattic/simplenote/InfoBottomSheetDialog.java) code. It didn't seem to cover the scenario when dialog fragment was dismissed by touching on the screen which is why I added `public void onDismiss(DialogInterface dialog)` callback. I'll test it further and see if we can eliminate one of them.

ashiagr replied:
Fixed here d15dd92

malinajirka commented on libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/widgets/WPBottomSheetDialogFragment.java:
```diff
@@ -0,0 +1,44 @@
+    @NonNull
+    @Override
+    public Dialog onCreateDialog(Bundle savedInstanceState) {
+        setRetainInstance(true);
```
Invoking `setRetainInstance(true)` on a fragment with UI is quite risky (we might/will leak all the views), do we really need it? Moreover, `SignupBottomSheetDialogFragment` currently holds a reference to the Activity, which will also leak after a configuration change. Wdyt?

ashiagr replied:
Fixed here 1d51b1f

ashiagr commented on libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/SignupBottomSheetDialogFragment.java:
```diff
@@ -0,0 +1,100 @@
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater,
```
I was trying to refer to [SimpleNote dialog](https://github.com/Automattic/simplenote-android/blob/6978afe6744db9c5f0a3141cccc1d3aa01c86772/Simplenote/src/main/java/com/automattic/simplenote/InfoBottomSheetDialog.java) - a dialog affected in Automattic/simplenote-android#792. while making the changes. Will consider your suggestion.

ashiagr replied:
Fixed here e0e8327

ashiagr commented on WordPress/src/main/java/org/wordpress/android/ui/accounts/LoginActivity.java:
```diff
@@ -165,8 +165,8 @@ protected void onCreate(Bundle savedInstanceState) {
             }
 
             if (mSignupSheetDisplayed) {
-                mSignupSheet = new SignupBottomSheetDialog(this, this);
```
I had tested it by removing this part. If dialog fragment was displayed, it wasn't shown again on orientation change. Will further test it out.

ashiagr replied:
Fixed here 8dcfbc7

ashiagr commented on libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/SignupBottomSheetDialogFragment.java:
```diff
@@ -0,0 +1,100 @@
+    public static final String TAG = SignupBottomSheetDialogFragment.class.getSimpleName();
+    private SignupSheetListener mSignupSheetListener;
+
+    public SignupBottomSheetDialogFragment(@NonNull final SignupSheetListener signupSheetListener) {
```
Fixed here 4fd8960

ashiagr commented:
Thank you for reviewing the changes @malinajirka 🙏! Fixed issues related to dialog to fragment conversion.

However this design issue still remains, 
> Emulator, Nexus 9, API 28
> the buttons are white on light background

Noticed it appears in SimpleNote android app as well 

<img width="867" alt="Screenshot 2020-01-14 at 5 42 44 PM" src="https://user-images.githubusercontent.com/1405144/72343616-4faf5700-36f5-11ea-925c-84188c73becf.png">

As we're using a light themed bottom sheet style, ideally navigation bar icons should have been darker in color. But it isn't the case here, Android platform provides little support in customising navigation bar icons color. @theck13 are you aware of the issue in SimpleNote app? Do you have any suggestions?

theck13 commented:
No, I wasn't aware of it.  No, I don't have any suggestions.

malinajirka commented:
Thanks @ashiagr! I've looked into this and I found what is causing the issue. It's the hack we use in `WPBottomSheetDialogFragment` to adjust the width of the view/window. Tbh I'd personally discuss this with a designer as I believe it's ok too keep the sheet fullwidth - it's the default OS behavior, right? If they think it's better to use a maxWidth, we'll need to come up with another solution - eg. different layout/dimens for tablets. Wdyt?

ashiagr commented:
Good catch @malinajirka ! Thanks for looking into it.

> I believe it's ok too keep the sheet fullwidth - it's the default OS behavior, right

Right, it is the default OS behavior. I agree we should keep it fullwidth. 

Screenshots for the fullwidth layout (for review):

| Landscape | Portrait | 
| ------ | ------------- | 
| ![device-2020-01-15-144701](https://user-images.githubusercontent.com/1405144/72425164-3c5fc280-37ad-11ea-9e14-a8ceb602e23b.png)| ![device-2020-01-15-144715](https://user-images.githubusercontent.com/1405144/72425182-45e92a80-37ad-11ea-8eca-bbb2a89f8284.png) |

> Besides I don't understand why we change the layout parameters of the window and not just parameters of the view.

Tried setting it on the fragment's view, it doesn't seem to be covering full width

![view](https://user-images.githubusercontent.com/1405144/72425039-00c4f880-37ad-11ea-9811-fa9c1ffb38cd.png)

It could be because full width setting for dialog  window `window.setLayout(-1, -1);` inside `BottomSheetDialog`. But we can investigate it more.

```
protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Window window = this.getWindow();
        if (window != null) {
            if (VERSION.SDK_INT >= 21) {
                window.clearFlags(67108864);
                window.addFlags(-2147483648);
            }
            window.setLayout(-1, -1);
        }
    }
```

theck13 commented:
I don't think we should settle for the full-width solution.  Restricting the bottom sheet to a maximum width is by design and has accessibility advantages as well.  I haven't seen another app using a full-width bottom sheet on a large screen (e.g. tablet in landscape orientation).  There are a few alternatives to leaving the bottom sheet full-width.
1. Make bottom sheet full-width _only for API 28 and large screen_.
2. Make style of navigation bar dark _only for API 28 and large screen_.
3. Find out why the navigation bar behavior is different on API 28 only.

ashiagr commented:
Thanks for great suggestions @theck13 ! 

> 1. Make bottom sheet full-width only for API 28 and large screen.
> 3. Find out why the navigation bar behavior is different on API 28 only.

One thing to note is white navigation bar buttons on light background issue is replicable on large screens/ tablets with version > v27 and not just v28. It means making large screen bottom sheet full-width for all API versions > v27. 

> Make style of navigation bar dark only for API 28 and large screen.

How about something like this? I changed nav bar background color slightly #`e9eff3`, so both the buttons and nav bar icons appear to be white on a light blue background. But they're not clearly visible. I'll need hex color value to which I should update nav bar background.

Also updated width to fix layout width issue.

| API | Portrait | Landscape |
| ------ | ------------- | ------------- |
|27, 29|![v29-portrait_400](https://user-images.githubusercontent.com/1405144/72718685-6e519a00-3b9c-11ea-9997-dae87f49556b.png)| ![v29-land](https://user-images.githubusercontent.com/1405144/72719030-18312680-3b9d-11ea-9d68-a97fe8dffcd8.png)|
|26| ![v26-portrait](https://user-images.githubusercontent.com/1405144/72719123-3dbe3000-3b9d-11ea-9e95-b0d0d2c16bb1.png)| ![v26-land](https://user-images.githubusercontent.com/1405144/72719154-4ca4e280-3b9d-11ea-9dd0-22ebf0eee317.png)|

Two more screenshots with dark nav bar background in different shades of gray (for design review).

| Dark Grey | Light Grey |
| ------------- | ------------- |
|![gray](https://user-images.githubusercontent.com/1405144/72720801-bd99c980-3ba0-11ea-8154-9aa7c3678196.png)| ![gray_light](https://user-images.githubusercontent.com/1405144/72721207-80820700-3ba1-11ea-9edc-f3535facea10.png)|

CC @osullivanchris

osullivanchris commented:
Hey @ashiagr 

There's a few things going on here so I might not have all the context but just one thing I don't think looks right is having the system navigation limited to the width of the sheet. Is that a limitation that we need to have, or a suggestion?

I actually think the images shown for API 26 and below look more like what I would expect. I was looking at the same issue [here](https://github.com/wordpress-mobile/WordPress-Android/pull/11107#issuecomment-576263999) and you can see that I'm just leaving the system bar unaffected by the sheet. 

Hope that makes sense. Let me know if there's more info I'm lacking so that I can better help

theck13 commented:
I don't think we can guarantee the width or location of the navigation bar button/icons.  As an example, navigation bar buttons/icons on the Pixel C are not centered and are much wider than the width of the bottom sheet.  Therefore, I don't think setting the `navigationBarColor` attribute to a dark color will help for all cases.  See the screenshots below for illustration.

<a href="https://user-images.githubusercontent.com/3827611/72748668-18b5c700-3b75-11ea-9dde-907080514c9c.png"><img src="https://user-images.githubusercontent.com/3827611/72748668-18b5c700-3b75-11ea-9dde-907080514c9c.png"></a>

ashiagr commented:
Thanks for taking a look @theck13 , @osullivanchris ! Managed to show full width navigation bar and restrict max width for tablets. Can you please review these screenshots?

**Tablet**

| API | Portrait | Landscape |
| ------ | ------------- | ------------- |
| 29 | ![v29_por](https://user-images.githubusercontent.com/1405144/72790129-fe561900-3c5a-11ea-966b-888ffeb1e43a.png)|  ![v29_land](https://user-images.githubusercontent.com/1405144/72790148-07df8100-3c5b-11ea-902e-c11201499470.png)|  
| 27 | ![v27_por](https://user-images.githubusercontent.com/1405144/72790178-1a59ba80-3c5b-11ea-939e-782922608778.png)| ![v27_land](https://user-images.githubusercontent.com/1405144/72790237-44ab7800-3c5b-11ea-9376-6188bcab5e28.png) |
| 26 | ![v26_portrait](https://user-images.githubusercontent.com/1405144/72790280-58ef7500-3c5b-11ea-81c8-b256991eb656.png)| ![v26_land](https://user-images.githubusercontent.com/1405144/72790297-5f7dec80-3c5b-11ea-8978-f9d452e3a59e.png)|

**Phone**

| API | Portrait | Landscape |
| ------ | ------------- | ------------- |
| 29 | ![v29_por](https://user-images.githubusercontent.com/1405144/72790833-50e40500-3c5c-11ea-9dc9-03a6094ef07b.png)|![v29_land](https://user-images.githubusercontent.com/1405144/72790858-593c4000-3c5c-11ea-8774-5d34a576bdc7.png) |
| 27 | ![v27_por](https://user-images.githubusercontent.com/1405144/72791018-9dc7db80-3c5c-11ea-9507-d267eb8d7327.png)| ![v27_land](https://user-images.githubusercontent.com/1405144/72791028-a3252600-3c5c-11ea-9f8f-3b01fa967bbf.png)|
| 26 | ![v26_por](https://user-images.githubusercontent.com/1405144/72791083-b801b980-3c5c-11ea-9dd7-3a4398d4c14b.png)|![v26_land](https://user-images.githubusercontent.com/1405144/72791177-dc5d9600-3c5c-11ea-86a5-187d7a1bfc3c.png) |

I will need 
1. max-width values for different tablet sizes in case default width `0dp (match_parent)` appears odd.
2. resource qualifiers that we use to differentiate tablets and phones. I find there's always confusion around it and I couldn't find this information [here](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/docs/using-android-resources.md).

osullivanchris commented:
> The buttons do looked a little squished on tablets in portrait and landscape orientation. We may want to add bottom_sheet_dialog_width to a few dimens.xml files. I defer to the designers on what values they would like for certain screen sizes and orientations.

Chatting to @SylvesterWilmott and he said we have a 512dp container that is used to limit the width of content which is used in various parts of the app, which is what we've done for Simplenote as well.

Will we use that as a starting point? We can add further definitions / breakpoints if it doesn't work in some context/usecase

ashiagr commented:
Thanks @osullivanchris ! 

As a starting point, I took screenshots for the bottom sheet on some of the popular tablets:
_Portrait Width_ `existing 300dp` 
_Landscape Width_ `512dp`

Let me know if we need to change the width on any of these and/or want me to test any other tablet. 

| Device/ Resolution/ Density/ API | Portrait | Landscape |
| ------|------------- |------------- |
| Pixel C/ 1280x900/ xhdpi/ 27 | ![PixelC_27_port_320](https://user-images.githubusercontent.com/1405144/73135560-84160200-4069-11ea-944b-bfe95cc14a39.png)| ![PixelC_27_land_512](https://user-images.githubusercontent.com/1405144/73135109-b5d89a00-4064-11ea-8fb3-5876726786d7.png)| 
| Nexus 10/ 1280x800/ xhdpi/ 27 |  ![nexus10_27_port_320dp](https://user-images.githubusercontent.com/1405144/73135566-9a23c280-4069-11ea-8fbe-b315431657f2.png) | ![nexus10_27_land_512dp](https://user-images.githubusercontent.com/1405144/73135139-f9cb9f00-4064-11ea-9b63-301b21f1e04a.png)|
| Nexus 9/ 1024x768/  xhdpi/ 29 | ![nexus9_29_port_320dp](https://user-images.githubusercontent.com/1405144/73135576-b45da080-4069-11ea-9196-b1e62b65c204.png)  | ![nexus9_29_land_512dp](https://user-images.githubusercontent.com/1405144/73135175-43b48500-4065-11ea-8a2d-7e58938e47d0.png)| 
| Samsung Galaxy Tab 10/ 800x1280/ mdpi/ 27 | ![samsung_27_port_320](https://user-images.githubusercontent.com/1405144/73135588-cf301500-4069-11ea-94dc-64ae5ae4db28.png) |    ![samsung_27_land_512](https://user-images.githubusercontent.com/1405144/73135195-70689c80-4065-11ea-83ed-c9dcc4df45b5.png)  | 
| Nexus 7 '13/ 600x960/ xhdpi/ 27 | ![nexus7_27_port_320dp_xhdpi](https://user-images.githubusercontent.com/1405144/73135798-7910a100-406c-11ea-9983-18e561cb07ef.png)| ![nexus7_27_land_512dp_xhdpi](https://user-images.githubusercontent.com/1405144/73135793-7150fc80-406c-11ea-8677-4decb68334ca.png)| 
| Nexus 7 '12/  600x960 / tvdpi/ 27|  ![nexus7_27_port_320dp_tvdpi](https://user-images.githubusercontent.com/1405144/73135749-eff96a00-406b-11ea-8334-b7c55cabbd64.png) |![nexus7_27_land_512dp_tvdpi](https://user-images.githubusercontent.com/1405144/73135805-96de0600-406c-11ea-9bdd-48f4a7b1a233.png)| 

@theck13 I see `bottom_sheet_dialog_width` for `sw600dp` configuration qualifier defined in WordPress resources: `<dimen name="bottom_sheet_dialog_width">300dp</dimen>`. 
Is it for a reason or can we move it to login library resources? I'll put corresponding `sw600dp-land` value accordingly.

theck13 commented:
> @theck13 I see `bottom_sheet_dialog_width` for `sw600dp` configuration qualifier defined in [WordPress resources](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/res/values-sw600dp/dimens.xml#L10). Is it for a reason or can we move it to login library resources? I'll put corresponding `sw600dp-land` value accordingly.

Since `WPBottomSheetDialogFragment` is a base class in a library, we don't want to specify certain dimensions because we don't know what the inherited class and layout will be in client apps.  The only value for `bottom_sheet_dialog_width` in the library is `0dp` so the container will wrap to the layout's content.  It's up to the client apps (e.g. WooCommerce and WordPress) to specify what the dimensions should be based on their desired layout.  That's why the values of `bottom_sheet_dialog_width` in `sw600dp` and `sw600dp-land` should be in WordPress and not WordPressLoginFlow.

ashiagr commented:
> If we find a scenario where this isn't working for us later I think we can add more breakpoints or definitions at that point. Would you agree?

I agree.

> do you think we should extend this to landscape on phones in any cases or keep it just for tablet?

I would want it to be consistent with our other apps like SimpleNote, and like to know @theck13's inputs.

theck13 commented:
Simplenote is different since it uses the `large` and `large-land` modifiers.

Personally, I think the bottom sheet looks a little stretched on phones in landscape orientation.  To adjust that and since the horizontal button layout only occurs in landscape orientation, we could add `bottom_sheet_dialog_width` with a value around `512dp` and the `w600dp-land` qualifier.  I haven't tried that out though so we'll want to test it to make sure it looks good.  Otherwise, it's looks good to me.

malinajirka commented on libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/widgets/WPBottomSheetDialogFragment.java:
```diff
@@ -0,0 +1,58 @@
+        // Limit width of bottom sheet on wide screens; non-zero width defined only for large qualifier.
+        if (dp > 0) {
+            FrameLayout bottomSheetLayout =
+                    dialog.findViewById(com.google.android.material.R.id.design_bottom_sheet);
```
>If that resource name is changed or removed, it can cause a NullPointerException crash from the bottomSheetLayout.getLayoutParams() call in the line below

We include the sources of the library in our APKs, so I believe it won't crash in runtime. 
We'll find out when we decide to update the version of the library that the resources is missing because the project won't build, right?
I agree that it's still considered an anti-pattern, but this whole customization of a component which isn't supposed to be customized is an anti-pattern. So I personally wouldn't worry about it too much unless we have an easy solution for it.
Wdyt?

theck13 replied:
If the resource is removed completely, it should cause a build error.  If the resource is moved to another view or layout, it would be a bug or crash.  The point is that we don't have control over the library resources.  Adding a `null`-check is just a safety measure.

malinajirka replied:
👍 yeah, adding null check is definitely a good call! Completely agree ;). Thanks for the elaboration:)

ashiagr commented:
Thanks @theck13 ! 

I tested `512dp` for `w600dp-land` on few phone emulators. @osullivanchris, can you review these screenshots?

| Device/ Resolution/ ViewPort/ Density | Landscape |
| ------|------------- |
| Nexus6p/ 1440x2560/ 412x738/ xxhdpi |  ![nexus6p](https://user-images.githubusercontent.com/1405144/73331478-e77e7a80-4289-11ea-884a-f01924ab0e66.png) |
| Pixel 3/ 1080x2160/ 411x823/ xxhdpi | ![pixel-3](https://user-images.githubusercontent.com/1405144/73331526-1399fb80-428a-11ea-80a4-ff423f86d170.png) |  
| Pixel 3 XL/ 1440x2960/ 411x846/ xxxhdpi | ![pixel-3xl](https://user-images.githubusercontent.com/1405144/73331539-244a7180-428a-11ea-8228-453225ab9d4c.png) |  
| Samsung S10/ 1440x3040/ 360x760/ xxxhdpi | ![samasung-s10](https://user-images.githubusercontent.com/1405144/73331561-388e6e80-428a-11ea-9501-6916e92f81f1.png) |  
| Galaxy Fold/ 840x1960/ /xhdpi  | ![samsung_galaxy_fold-7 3_inch](https://user-images.githubusercontent.com/1405144/73331594-49d77b00-428a-11ea-8efb-cbc4d5eb166e.png) |

theck13 commented:
Now that we have `bottom_sheet_dialog_width` with `512dp` for `sw600dp-land` and `w600dp-land`, I think we can delete the `sw600dp-land` instance since it's redundant.  @ashiagr and @malinajirka, correct me if I'm wrong or I'm forgetting a scenario for which we need to keep `sw600dp-land`.

ashiagr commented:
Thanks for the suggestion @theck13 . I removed the value from `sw600dp-land` and tested it on a Nexus 10 tablet emulator, buttons looked squished again.

![nexus_10_landscape](https://user-images.githubusercontent.com/1405144/73520089-c3a56b00-4428-11ea-8c40-e80d45d62957.png)

According to [Android's best-matching resource guide](https://developer.android.com/guide/topics/resources/providing-resources#BestMatch) , it seems we have this order of precedence:

`values-sw600dp-land`
`values-sw600dp`
`values-w600dp-land`

If we remove `values-sw600dp-land`,  it picks up next precedence which is `values-sw600dp` - `300dp` for the tablet.

ashiagr commented:
@theck13 do you find the solution good enough to be merged? If yes, can you approve these changes?

theck13 commented:
I haven't found a better solution, but I haven't been researching this either.  Therefore, we're going to use this is Simplenote (https://github.com/Automattic/simplenote-android/pull/933).  I haven't reviewed the code as extensively as @malinajirka and he had requested changes in his review.  So I'll leave the final approval to him.

malinajirka commented:
Thanks @ashiagr for all the improvements!! ;) The final solution is great.

I've been testing the sign-up flow and I noticed two bugs.
1. The first issue is already being tracked so it apparently isn't related your changes. I'm just mentioning it here to keep thinks linked. https://github.com/wordpress-mobile/WordPress-Android/issues/11242

2. The bottom sheet doesn't disappear after rotation.
- Click on "SIGN UP FOR WORDPRESS.COM"
- Rotate your device
- Click on "SIGN UP WITH EMAIL"
- notice the bottom sheet doesn't disappear (Do we perhaps show two dialogs on top of one another?)
<img src="https://user-images.githubusercontent.com/2261188/74147635-ec391c00-4c03-11ea-8821-22a94a7f9530.gif" width="300px" />

jkmassel commented:
We're freezing `14.2` today, so this PR is being bumped to `14.3`. If you'd like it to ship with `14.2`, please merge it into `release/14.2` and ping me – I'll be happy to cut a new beta release.

ashiagr commented:
>  2. The bottom sheet doesn't disappear after rotation.

Thanks for catching the bug! Fixed here f963bc2

malinajirka commented:
@theck13 It seems, I can't merge the PR, since you requested some changes. Could you please either dismiss the review or approve it and merge the PR. Thanks!
